pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -104,20 +96,41 @@ private SchemaType getSchemaTypeOrDefault(String topic, Class<?> clazz) {
     }
 
     private static SchemaType getDefaultSchemaType(Class<?> clazz) {
-        if (byte[].class.equals(clazz)
-            || ByteBuf.class.equals(clazz)
-            || ByteBuffer.class.equals(clazz)) {
-            return SchemaType.NONE;
+        if (byte[].class.equals(clazz)) {
```

Comment:
for bytes for example, we are moving away from using None schema to Bytes. This change should be backwards compatible since the schemas themselves are.
The reason for switching is that I believe the schema integration should rely on the decisions of Schema definitions rather than taking those decisions here.